src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderItemsController.cs

```diff
@@ -118,5 +118,36 @@ public async Task<ActionResult<GetOrderItemModel>> GetAsync(CallOffId callOffId,
 
             return ValidationProblem(ModelState);
         }
+
+        [Authorize(Policy = PolicyName.CanManageOrders)]
+        [HttpDelete("{catalogueItemId}")]
+        public async Task<ActionResult> DeleteOrderItemAsync(
+            CallOffId callOffId,
+            CatalogueItemId catalogueItemId)
+        {
+            Expression<Func<Order, IEnumerable<OrderItem>>> orderItems = o =>
+                o.OrderItems.Where(i => i.CatalogueItem.Id == catalogueItemId);
+
+            var order = await context.Order
+                .Where(o => o.Id == callOffId.Id)
+                .Include(orderItems)
+                .ThenInclude(x => x.CatalogueItem)
+                .SingleOrDefaultAsync();
+
+            if (order is null || order.IsDeleted)
+            {
+                return NotFound();
+            }
+
+            var deleted = order.DeleteOrderItem(catalogueItemId);
+            if (!deleted)
+            {
+                return NotFound();
+            }
```

Comment:
I'm not sure we gain anything by introducing the variable here.

Reply:
done